test/unit.test.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const pino = require('pino')
+
+const proxyquire = require('proxyquire')
+
+const test = require('tap').test
+const options = {
+  index: 'pinotest',
+  type: 'log',
+  consistency: 'one',
+  host: 'localhost',
+  port: 9200
+}
+test('make sure log is a valid json', (t) => {
+  t.plan(2)
+  const Client = function (config) {
+    t.equal(config.host, `${options.host}:${options.port}`)
+  }
+  Client.prototype.index = (obj, cb) => {
+    t.ok(obj, true)
+    cb(null, {})
+  }
+  // TODO: test it better
+  // Client.prototype.bulk = (obj, cb) => {
+  //   cb(null, {})
+  // }
```

Comment:
I found it hard to test because of the code use a global reference to 'splitter'.
`writev` -> `client.bulk` -> `splitter.emit('insert', create, chunks[i].chunk)`
and 
`write` -> `client.index` -> `splitter.emit('insert', data, body)`

I thought about breaking the splitter to it's own file this way I can test this piece by itself